### PR TITLE
Enable WebTransport over H3 server for Firefox nightly

### LINKED ISSUE #39309: WebTransport available in Firefox Nightly 113
WebTransport is now [enabled in Nightly-only builds](https://searchfox.org/mozilla-central/source/modules/libpref/init/StaticPrefList.yaml#12527,12529) as of Firefox 113. 🎉

But https://wpt.fyi/results/webtransport?label=experimental&label=master&aligned showing SKIP for Firefox, not green.

When I _"Run in your browser on wpt.live"_ e.g. https://wpt.live/webtransport/connect.https.any.html in Chrome, the tests run immediately (3 pass, 4 fail since there's no server when invoked this way). But in Firefox the tests never run, the page stalls for a few seconds and then times out.

Looks like some infrastructure needs updating.

### PR DESCRIPTION
**Pull request by bashi**
Fix #39309.

### CODE REVIEW

**Review by gsnedders — COMMENTED** [review]
This looks good to me, but given it changes the default status for Firefox @jgraham should probably review this.

**Review by jan-ivar — APPROVED** [review]

**Comment by jesup** [comment]
This looks good to me (I've been running wpts locally in Mozilla's CI).   jgraham is out, but he knows that we're enabling this; I've been working with him.  Let's merge it

File: tools/wpt/run.py [hunk 1]
```diff
@@ -265,6 +265,9 @@ def setup_kwargs(self, kwargs):
             kwargs["headless"] = True
             logger.info("Running in headless mode, pass --no-headless to disable")
 
+        if kwargs["browser_channel"] == "nightly" and kwargs["enable_webtransport_h3"] is None:
```
Comment (jgraham) [inline tools/wpt/run.py:268]: Putting this in the frontend seems a bit strange; I guess ideally we want this to be a property of the product. But that's a slightly bigger refactor, so I guess it's fine to do this for now.

**Review by jgraham — CHANGES_REQUESTED** [review]
It has also been pointed out to me by @jesup that we have `ini` files checked in with the tests that are being used to disable all the actual tests in !chrome. Although that works it's quite unexpected; we at least need to update those files, but we also need to consider if we can just remove the ini files and let the tests fail in browsers without the webtransport server.

**Comment by jan-ivar** [comment]
> ... let the tests fail in browsers without the webtransport server.

Why not just run the server for these tests regardless of browser?

**Comment by bashi** [comment]
At the time the WebTransport server was introduced, test infra for browsers other than chrome weren't ready to run the server, mostly due to aioquic dependency (IIUC). I'm not sure the current situation but probably we can run the server on all platforms?

**Comment by jgraham** [comment]
Yeah, the binary dependency is what makes this tricky; in Gecko CI we've had to add aioquic to our PyPI mirror for this to work, and it's still pretty fragile if we update the version.

**Review by jgraham — APPROVED** [review]

**Pull request merged**